Objective-C/CBLReplicatorConfiguration.m

```diff
@@ -232,9 +264,8 @@ - (void) addCollection: (CBLCollection*)collection
         }
     }
     
-    CBLCollection* defaultCollection = [_database defaultCollectionOrThrow];
-    if (collection == defaultCollection)
-        [self checkAndUpdateConfig: config];
+    if (!_database)
+        _database = collection.db;
```

Comment:
I think it would be more safe to also check if _database is nil or not. If it's not nil, throw an exception as well.

Reply:
but we are not setting the _database  property in case of initWithTarget:, we link it when user calls the addCollection:config

Reply:
The assumption is that if there is a collection added, the database cannot be nil. As collection.db is weak, there is no guarantee that it will not be null even after calling collection.isValid() (technically). 

Also, I just noticed that _collections is only populated when collection property is called. I think we actually do not need _collections at all.

Also, the below logic is not correct. I should check against _database if it is not nil. If it's nil, it will just set _database.

```
if (_collections.count > 0) {
        if (_collections[0].db != collection.db) {
            [NSException raise: NSInvalidArgumentException
                        format: @"Attempt to add collection from different databases"];
        }
    }
    
    if (!_database)
        _database = collection.db;
```

Reply:
simplified